### PR TITLE
Fixes camera taking pictures when you try to put them on a table

### LINKED ISSUE #12121: Cameras take pictures when you place them on a table
It also happens if you put them in a box.

### PR DESCRIPTION
**Pull request by JJRcop**
Also fixes this when trying to put them in a container

Fixes #12121

### CODE REVIEW

**Comment by PKPenguin321** [comment]
nice :+1:

File: code/modules/paperwork/photography.dm [hunk 1]
```diff
@@ -416,7 +416,12 @@
 		viewpichelper(Ainfo)
 
 /obj/item/device/camera/afterattack(atom/target, mob/user, flag)
-	if(!on || !pictures_left || ismob(target.loc)) return
+	if(!on || !pictures_left || ismob(target.loc) || !isturf(target.loc)) return
+	if(user.Adjacent(target))
+		var/list/bad_targets = list(/obj/structure, /obj/item/weapon/storage)
+		for(var/T in bad_targets)
```
Comment (RemieRichards) [inline code/modules/paperwork/photography.dm]: ```is_type_in_list()``` please

Comment (JJRcop) [inline code/modules/paperwork/photography.dm]: oh, neat.

File: code/modules/paperwork/photography.dm [hunk 2]
```diff
@@ -416,7 +416,11 @@
 		viewpichelper(Ainfo)
 
 /obj/item/device/camera/afterattack(atom/target, mob/user, flag)
-	if(!on || !pictures_left || ismob(target.loc)) return
+	if(!on || !pictures_left || ismob(target.loc) || !isturf(target.loc)) return
```
Comment (RemieRichards) [inline code/modules/paperwork/photography.dm]: move these ```return```s to different lines

File: code/modules/paperwork/photography.dm [hunk 3]
```diff
@@ -416,7 +416,11 @@
 		viewpichelper(Ainfo)
 
 /obj/item/device/camera/afterattack(atom/target, mob/user, flag)
-	if(!on || !pictures_left || ismob(target.loc)) return
+	if(!on || !pictures_left || ismob(target.loc) || !isturf(target.loc)) return
+	if(user.Adjacent(target))
+		var/list/bad_targets = list(/obj/structure, /obj/item/weapon/storage)
+		if(is_type_in_list(target, bad_targets)) return
```
Comment (RemieRichards) [inline code/modules/paperwork/photography.dm]: move these ```return```s to different lines

**Comment by phil235** [comment]
:+1:

**Pull request merged**